omniforms/wagtail/forms.py

```diff
@@ -0,0 +1,175 @@
+        """
+        Django properties
+        """
+        model = Group
+        fields = ("permissions",)
+        widgets = {'permissions': forms.CheckboxSelectMultiple()}
+
+    @staticmethod
+    def _checkboxes_by_id(bound_field):
+        """
+        Generates and returns a dict of checkbox fields keyed by permission ID
+
+        :param bound_field: The bound form field to get permission checkboxes from
+        :return: Dict of checkboxes keyed by permission ID
+        """
+        if django.VERSION < (1, 11):
+            return {int(checkbox.choice_value): checkbox for checkbox in bound_field}
+        return {int(checkbox.data['value']): checkbox for checkbox in bound_field}
+
+    def as_admin_panel(self):
+        """
+        Required by the wagtail so that it can render the panel html when this
+        form is registered using the register_group_permission_panel hook
+
+        This code is basically taken from wagtails own 'format_permissions' template
+        tag and modified/reduced to work with our specific use case
+
+        :return: Rendered form panel
+        """
+        permissions = self.fields['permissions'].queryset
+        checkboxes_by_id = self._checkboxes_by_id(self['permissions'])
+        object_perms = []
+
+        for content_type in ContentType.objects.filter(permission__in=permissions).distinct():
+            content_perms_dict = {'object': content_type.name}
+            for perm in permissions.filter(content_type=content_type):
+                permission_action = perm.codename.split('_')[0]
+                if permission_action in ['add', 'change', 'delete']:
+                    content_perms_dict[permission_action] = checkboxes_by_id[perm.id]
+
+            if content_perms_dict:
+                object_perms.append(content_perms_dict)
+
+        return render_to_string(
+            'modeladmin/omniforms/wagtail/includes/permissions.html',
+            {'title': self.admin_panel_title, 'object_perms': object_perms}
+        )
+
+    def save(self, commit=True):
+        """
+        We need to make sure we preserve permissions which are not managed by this form
+        as such we get a queryset of all permissions associated with the forms model instance
+        before calling super so that we may re-associate them with the instance after the form
+        has been saved
+
+        :param commit: Whether or not to commit the changes to the database
+        :return: The saved group instance
+        """
+        # We go back to the object to read (in order to reapply) the
+        # permissions which were set on this group, but which are not
+        # accessible in the wagtail admin interface, as otherwise these would
+        # be clobbered by this form.
+        try:
+            untouchable_permissions = self.instance.permissions.exclude(pk__in=self.fields['permissions'].queryset)
+            bool(untouchable_permissions)  # force this to be evaluated, as it's about to change
```

Comment:
Potentially cast to a list, add to comment the reasoning (the queryset is lazy).

Reply:
Done in https://github.com/omni-digital/omni-forms/pull/36/commits/616478d79f0834b625d8ab28e7b0e1b3d85b6ee3